### PR TITLE
DEV-1763: Report os arch, family and kernel from client on connect.

### PR DESCRIPTION
**Pull request by matematik7**
Report os arch, family and kernel from client on connect.

Adding this to sessions api endpoint will be done in next pr.

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -375,3 +326,78 @@ func (c *Client) setCurCmdPID(pid *int) {
...
+		ID:      c.config.Client.ID,
+		Name:    c.config.Client.Name,
+		OSArch:  c.systemInfo.GoArch(),
+		Tags:    c.config.Client.Tags,
+		Remotes: c.config.Client.remotes,
+	}
+
+	info, err := c.systemInfo.HostInfo(ctx)
+	if err != nil {
+		c.Logger.Errorf("Could not get os information: %v", err)
```
Comment (m-terel) [inline client/client.go:345]: In the ticket it's listed `the clients must report their os family, os kernel, and the os arch`. Is all this info not required to fill-in and is it expected just to log the error and to proceed with the execution? If so, please add a comment.

File: client/client.go [hunk 2]
```diff
@@ -190,10 +180,7 @@ func (c *Client) connectionLoop(ctx context.Context) {
 			c.Errorf(err.Error())
 			break
 		}
-		ipv4, ipv6, _ := localIPAddresses()
-		c.connReq.IPv4 = ipv4
-		c.connReq.IPv6 = ipv6
-		req, _ := chshare.EncodeConnectionRequest(c.connReq)
+		req, _ := chshare.EncodeConnectionRequest(c.connectionRequest(ctx))
```
[the remark this answers is not in the document]
Comment (matematik7) [reply client/client.go]: Done

File: client/client.go [hunk 3]
```diff
@@ -375,3 +326,78 @@ func (c *Client) setCurCmdPID(pid *int) {
...
+	if info == nil {
+		return "", nil
+	} else if info.OS == "windows" {
+		return info.Platform + " " + info.PlatformVersion + " " + info.PlatformFamily, nil
+	}
+	return c.systemInfo.Uname(ctx)
+}
+
+// returns all local ipv4, ipv6 addresses
+func (c *Client) localIPAddresses() ([]string, []string, error) {
```
[the remark this answers is not in the document]
Comment (matematik7) [reply client/client.go]: I thought it made sense to move it below together with other funcs, but sure.

File: client/client.go (hunk 1 above)
Comment (matematik7) [reply client/client.go:378]: I chatted with Thorsten, and the client should report `unknown` in case of error, so that's what I changed it to.

**Review by m-terel — APPROVED** [review]

**Pull request merged**